core/src/main/java/org/wso2/carbon/kernel/utils/ConfigUtil.java

```diff
@@ -0,0 +1,613 @@
+                    CONFIG_FILE_NAME, inputString);
+        } else {
+            msg = String.format("Unsupported placeholder type: %s", type.getValue());
+        }
+        logger.error(msg);
+        throw new RuntimeException(msg);
+    }
+
+    /**
+     * This method will concatenate and return the file types that need a root element. File types will be separated
+     * by | token. This method will be used to create the {@link ConfigUtil#FILE_REGEX}.
+     *
+     * @return String that contains file types which are separated by | token.
+     */
+    private static String getFileTypesString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (FileType fileType : FileType.values()) {
+            stringBuilder.append(fileType.getValue()).append("|");
+        }
+        String value = stringBuilder.toString();
+        logger.debug("FileTypes String: {}", value);
+        return value.substring(0, value.length() - 1);
+    }
+
+    /**
+     * This method will concatenate and return the placeholder types.. Placeholder types will be separated
+     * by | token. This method will be used to create the {@link ConfigUtil#PLACEHOLDER_REGEX}.
+     *
+     * @return String that contains placeholder types which are separated by | token.
+     */
+    private static String getPlaceholderString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (Placeholder placeholder : Placeholder.values()) {
+            stringBuilder.append(placeholder.getValue()).append("|");
+        }
+        String value = stringBuilder.toString();
+        logger.debug("PlaceHolders String: {}", value);
+        return value.substring(0, value.length() - 1);
+    }
+
+    /**
+     * This method will return the {@link ConfigUtil#CONFIG_FILE_NAME} file path. Path will be read using
+     * {@link ConfigUtil#FILE_PATH_KEY} environment variable or {@link ConfigUtil#FILE_PATH_KEY}
+     * system variable respectively. If both are null, CARBON_HOME/conf will be set as the default location.
+     *
+     * @return Location of the {@link ConfigUtil#CONFIG_FILE_NAME} file.
+     */
+    private static String getPath() {
+        String fileLocation = System.getenv(FILE_PATH_KEY);
+        if (!StringUtils.isNullOrEmpty(fileLocation)) {
+            logger.debug("{} location found in system variables. Location: {}", FILE_PATH_KEY, fileLocation);
+            return fileLocation;
+        }
+        if (fileLocation != null && fileLocation.isEmpty()) {
+            logger.warn("{} found in environment variables. But value is empty.", FILE_PATH_KEY);
+        }
+        fileLocation = System.getProperty(FILE_PATH_KEY);
+        if (!StringUtils.isNullOrEmpty(fileLocation)) {
+            logger.debug("{} location found in environment variables. Location: {}", FILE_PATH_KEY, fileLocation);
+            return fileLocation;
+        }
+        if (fileLocation != null && fileLocation.isEmpty()) {
```

Comment:
This is done to to simplify the code logic. If the "if " condition is satisfied, we can return the value. So if the next "if" block is reached, by default it implies else.